input/postgres/statements.go

```diff
@@ -131,10 +137,14 @@ func GetStatements(server *state.Server, logger *util.Logger, db *sql.DB, global
 		var e *pq.Error
 		if !usingStatsHelper && errors.As(err, &e) && (e.Code == "42P01" || e.Code == "42883") { // undefined_table / undefined_function
 			var pgssSchema string
-			err = db.QueryRow("SELECT nspname FROM pg_extension pge INNER JOIN pg_namespace pgn ON pge.extnamespace = pgn.oid WHERE pge.extname = 'pg_stat_statements'").Scan(&pgssSchema)
+			var foundExtVersion float32
+			err = db.QueryRow("SELECT nspname, extversion FROM pg_extension pge INNER JOIN pg_namespace pgn ON pge.extnamespace = pgn.oid WHERE pge.extname = 'pg_stat_statements'").Scan(&pgssSchema, &foundExtVersion)
 			if err == nil && pgssSchema != "public" {
 				return nil, nil, nil, fmt.Errorf("pg_stat_statements must be created in schema \"public\"; found in schema \"%s\"", pgssSchema)
 			}
+			if err == nil && extVersion != foundExtVersion && foundExtVersion <= maxKnownExtVersion {
```

Comment:
Ah, so if there's a version we don't know about but might work, we just keep our fingers crossed and go on, right?

Separately, I wonder if requiring the latest version for older Postgres versions could be a breaking change here. E.g., if currently we have some customers on say 1.7 for pg13 for whatever reason, but everything works fine because 1.7 is identical to 1.8 in terms of what we care about? I don't know whether this is theoretically even possible (i.e., I don't know if there are released versions of the extension with lower version numbers but still-compatible-for-us APIs), and even if so, whether it's a problem in practice, but maybe something to consider.

Reply:
Hm, that's a good point - it could be better to not error out here unnecessarily if the schema is compatible.

Possibly go with the approach where we decide which pg_stat_statements schema to expect based on the schema version, so we can even work without the extension upgrade?

Reply:
Sorry for the delay on this PR. @lfittl this is ready for review. I went ahead and added the schema detection code we discussed.